lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.mel

```diff
@@ -0,0 +1,640 @@
+// Verbatim copy of fileOptions.mel:pushOptionsUITemplate(), which is not
```

Comment:
Main file in the review.  Unfortunately mostly copy-pasted and adapted from Maya code, which was not designed to be extensible to use cases like Maya reference prim creation.

Reply:
I'm honestly not very knowledgeable about MEL and always struggle to write it, so I'm not going to pretend to review it. This being mostly UI code, I'll approve it blindly.